test/powershell/Language/Operators/SplitOperator.Tests.ps1

```diff
@@ -0,0 +1,236 @@
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b c d"
+
+            $res = "a b c d" -split " ", 0
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+
+            $res = "a b c d" -split " ", -1
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+        }
+
+        It "Binary split operator can works with freeform delimiter" {
+            $res = "a::b::c::d" -split "::"
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+        }
+
+        It "Binary split operator can preserve delimiter" {
+            $res = "a1:b1:c1:d" -split "(1:)"
+            $res.count | Should Be 7
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "1:"
+            $res[2] | Should Be "b"
+            $res[3] | Should Be "1:"
+            $res[4] | Should Be "c"
+            $res[5] | Should Be "1:"
+            $res[6] | Should Be "d"
+
+            $res = "a1:b1:c1:d" -split "1(:)"
+            $res.count | Should Be 7
+            $res[0] | Should Be "a"
+            $res[1] | Should Be ":"
+            $res[2] | Should Be "b"
+            $res[3] | Should Be ":"
+            $res[4] | Should Be "c"
+            $res[5] | Should Be ":"
+            $res[6] | Should Be "d"
+        }
+
+        It "Binary split operator can be case-insensitive and case-sensitive" {
+            $res = "abcBd" -split "B"
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+            $res[2] | Should Be "d"
+
+            $res = "abcBd" -isplit "B"
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+            $res[2] | Should Be "d"
+
+            $res = "abcBd" -csplit "B"
+            $res.count | Should Be 2
+            $res[0] | Should Be "abc"
+            $res[1] | Should Be "d"
+
+            $res = "abcBd" -csplit "B", 0 , 'IgnoreCase'
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+            $res[2] | Should Be "d"
+        }
+
+        It "Binary split operator can works with script block" {
+            $res = "a::b::c::d" -split {$_ -eq "b" -or $_ -eq "C"}
+            $res.count | Should Be 3
+            $res[0] | Should Be "a::"
+            $res[1] | Should Be "::"
+            $res[2] | Should Be "::d"
+        }
+
+    }
+
+    Context "Binary split operator options" {
+        BeforeAll {
+            $testText = "a12a`nb34b`nc56c`nd78d"
+            # Add '#' - now second line do not start with 'b'.
```

Comment:
let's use some other char instead of `#` - it's easy to confuse with comments, but comments are irrelevant here. 

also nit: `doesn't start`

Reply:
Fixed.